### PR TITLE
Better graphing

### PR DESCRIPTION
**Pull request by lustefaniak**

### CODE REVIEW

File: frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if (scheduler.isEmpty) {
+      val timer: SetTimeoutHandle = scala.scalajs.js.timers.setTimeout(updateFrequency) {
+        scheduler = js.undefined
+        applyGraphOperations()
+      }
+      scheduler = timer
+    }
+  }
+
+  private[this] var operationsToApply = js.Array[T]()
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703397/issues/source?bid=3121282&fileBranchId=3158877#l19)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703397/issues/source?bid=3121282&fileBranchId=3158877#l19)

File: frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package akkaviz.frontend
+
+import scala.concurrent.duration._
+import scala.scalajs.js
+import scala.scalajs.js.timers.SetTimeoutHandle
+
+class ScheduledQueue[T](fn: js.Array[T] => Unit, updateFrequency: FiniteDuration = 500.millis) {
+  private[this] var scheduler: js.UndefOr[SetTimeoutHandle] = js.undefined
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703397/issues/source?bid=3121282&fileBranchId=3158877#l8)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703397/issues/source?bid=3121282&fileBranchId=3158877#l8)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 3]
```diff
@@ -1,81 +1,86 @@
...
 class GraphView(showUnconnected: Var[Boolean]) extends Component {
 
-  override def render: Element = {
-
-    //FIXME: port initialization of graph from JS to scala and do it here
-    js.undefined.asInstanceOf
+  private[this] lazy val visibleNodes = js.Dictionary[Unit]()
+  private[this] lazy val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] lazy val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] var network: js.UndefOr[vis.Network] = js.undefined
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703402/issues/source?bid=3121282&fileBranchId=3158877#l17)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703402/issues/source?bid=3121282&fileBranchId=3158877#l17)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 4]
```diff
@@ -1,81 +1,86 @@
...
-  private def scheduleGraphOperations(): Unit = {
-    if (scheduler.isEmpty) {
-      val timer: SetTimeoutHandle = scala.scalajs.js.timers.setTimeout(updateFrequency) {
-        scheduler = js.undefined
-        applyGraphOperations()
-      }
-      scheduler = timer
-    }
-  }
+  private def applyGraphOperations(operationsToApply: js.Array[GraphView.GraphOperation]): Unit = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [applyGraphOperations should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703402/issues/source?bid=3121282&fileBranchId=3158877#l56)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [applyGraphOperations should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703402/issues/source?bid=3121282&fileBranchId=3158877#l56)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  object Edge {
+    def apply(id: String, from: String, to: String, arrows: js.UndefOr[String] = "to", title: js.UndefOr[String] = js.undefined, value: js.UndefOr[Int] = js.undefined): vis.Edge = {
+      js.Dynamic.literal(id = id, from = from, to = to, value = value, arrows = arrows).asInstanceOf[vis.Edge]
+    }
+  }
+
+  @js.native
+  trait DataSetOptions extends js.Any {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Redundant braces after class definition](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l38)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Redundant braces after class definition](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l38)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+    def get(d: js.Array[String]): js.Array[T] = js.native
+
+    def setOptions(o: DataSetOptions): Unit = js.native
+
+  }
+
+  @js.native
+  trait NetworkData extends js.Any {
+    var nodes: js.Array[Node] | DataSet[Node] = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l64)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l64)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    var edges: js.Array[Edge] | DataSet[Edge] = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l65)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l65)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    }
+  }
+
+  @js.native
+  trait Edge extends js.Any {
+    def id: js.Any
+  }
+
+  object Edge {
+    def apply(id: String, from: String, to: String, arrows: js.UndefOr[String] = "to", title: js.UndefOr[String] = js.undefined, value: js.UndefOr[Int] = js.undefined): vis.Edge = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [File line length exceeds 160 characters](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l32)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [File line length exceeds 160 characters](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l32)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    }
+  }
+
+  @js.native
+  @JSName("vis.Network")
+  class Network(element: Element, data: NetworkData, options: NetworkOptions) extends js.Any {
+
+    def destroy(): Unit = js.native
+
+    def fit() = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l130)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l130)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  object NetworkData {
+    def apply(nodes: js.Array[Node] | DataSet[Node], edges: js.Array[Edge] | DataSet[Edge]): NetworkData = {
+      js.Dynamic.literal(nodes = nodes.asInstanceOf[js.Any], edges = edges.asInstanceOf[js.Any]).asInstanceOf[vis.NetworkData]
+    }
+  }
+
+  @js.native
+  trait NetworkOptions extends js.Any {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Redundant braces after class definition](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l75)

Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Redundant braces after class definition](https://www.codacy.com/app/lustefaniak/akka-viz/file/2155703419/issues/source?bid=3121282&fileBranchId=3158877#l75)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 11]
```diff
@@ -1,81 +1,154 @@
...
+import scala.scalajs.js.|
+import scala.scalajs.js.|._
 
 class GraphView(showUnconnected: Var[Boolean]) extends Component {
 
-  override def render: Element = {
+  private[this] lazy val visibleNodes = js.Dictionary[Unit]()
+  private[this] lazy val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] lazy val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] var network: js.UndefOr[vis.Network] = js.undefined
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111825/issues/source?bid=3121282&fileBranchId=3158877#l18)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+    def add(d: T): js.Array[String] = js.native
+
+    def add(d: js.Array[T]): js.Array[String] = js.native
+
+    def remove(d: String): js.Array[String] = js.native
+
+    def remove(d: js.Array[String]): js.Array[String] = js.native
+
+    def get(d: String): T = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l52)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+  @js.native
+  @JSName("vis.Network")
+  class Network(element: Element, data: NetworkData, options: NetworkOptions) extends js.Any {
+
+    //Global methods for the network
+    def destroy(): Unit = js.native
+
+    def setData(data: NetworkData): Unit = js.native
+
+    def setOptions(options: NetworkOptions): Unit = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l120)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+    def once(event: String, fn: js.Function): Unit = js.native
+
+    //Methods related to the canvas
+    def canvasToDOM(coords: Coords): Coords = js.native
+
+    def DOMtoCanvas(coords: Coords): Coords = js.native
+
+    def redraw(): Unit = js.native
+
+    def setSize(width: String, height: String): Unit = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l135)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+  }
+
+  @js.native
+  @JSName("vis.Network")
+  class Network(element: Element, data: NetworkData, options: NetworkOptions) extends js.Any {
+
+    //Global methods for the network
+    def destroy(): Unit = js.native
+
+    def setData(data: NetworkData): Unit = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l118)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 16]
```diff
@@ -0,0 +1,249 @@
...
+    def setSize(width: String, height: String): Unit = js.native
+
+    //Methods to get information on nodes and edges.
+
+    //Physics methods to control when the simulation should run.
+
+    //Selection methods for nodes and edges.
+
+    //Methods to control the viewport for zoom and animation.
+    def getScale(): Double = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l144)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 17]
```diff
@@ -0,0 +1,249 @@
...
+
+    def remove(d: String): js.Array[String] = js.native
+
+    def remove(d: js.Array[String]): js.Array[String] = js.native
+
+    def get(d: String): T = js.native
+
+    def get(d: js.Array[String]): js.Array[T] = js.native
+
+    def setOptions(o: DataSetOptions): Unit = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l56)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 18]
```diff
@@ -0,0 +1,249 @@
...
+    //Methods to get information on nodes and edges.
+
+    //Physics methods to control when the simulation should run.
+
+    //Selection methods for nodes and edges.
+
+    //Methods to control the viewport for zoom and animation.
+    def getScale(): Double = js.native
+
+    def getViewPosition(): Coords = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l146)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 19]
```diff
@@ -0,0 +1,249 @@
...
+
+    def add(d: js.Array[T]): js.Array[String] = js.native
+
+    def remove(d: String): js.Array[String] = js.native
+
+    def remove(d: js.Array[String]): js.Array[String] = js.native
+
+    def get(d: String): T = js.native
+
+    def get(d: js.Array[String]): js.Array[T] = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l54)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 20]
```diff
@@ -0,0 +1,249 @@
...
+  }
+
+  @js.native
+  trait ZoomEvent extends js.Any {
+    var direction: String = js.native
+    var scale: Double = js.native
+  }
+
+  implicit class NetworkStaticEvents(val underlying: Network) extends AnyVal {
+    def onClick(fn: ClickEvent => Unit) = underlying.on("click", fn)
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/lustefaniak/akka-viz/file/2159111834/issues/source?bid=3121282&fileBranchId=3158877#l200)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 21]
```diff
@@ -1,81 +1,159 @@
...
+import scala.scalajs.js.|._
 
-class GraphView(showUnconnected: Var[Boolean]) extends Component {
+class GraphView(showUnconnected: Var[Boolean], actorSelectionToggler: (String) => Unit) extends Component {
 
-  override def render: Element = {
+  private[this] lazy val visibleNodes = js.Dictionary[Unit]()
+  private[this] lazy val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] lazy val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] var network: js.UndefOr[vis.Network] = js.undefined
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736317/issues/source?bid=3121282&fileBranchId=3158877#l17)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 22]
```diff
@@ -1,81 +1,159 @@
...
+      (event: vis.ClickEvent) =>
+        event.nodes.foreach(actorSelectionToggler)
+    }
 
-  private val updateFrequency: FiniteDuration = 500.millis
+    network = n
+  }
 
-  private var scheduler: js.UndefOr[SetTimeoutHandle] = js.undefined
+  private def graphSettings: NetworkOptions = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [graphSettings should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736317/issues/source?bid=3121282&fileBranchId=3158877#l32)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 23]
```diff
@@ -0,0 +1,282 @@
...
+
+  object Coords {
+    def apply(x: Double, y: Double): Coords = {
+      js.Dynamic.literal(x = x, y = y).asInstanceOf[Coords]
+    }
+  }
+
+  @js.native
+  trait NetworkMoveToOptions extends js.Any {
+    var position: Coords = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l96)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 24]
```diff
@@ -0,0 +1,282 @@
...
+    def onShowPopup(fn: String => Unit) = underlying.on("showPopup", fn)
+
+    def onHidePopup(fn: () => Unit) = underlying.on("hidePopup", fn)
+
+    //Events triggered by the rendering module. Can be used to draw custom elements on the canvas.
+    def onInitRedraw(fn: () => Unit) = underlying.on("initRedraw", fn)
+
+    def onBeforeDrawing(fn: (Canvas) => Unit) = underlying.on("beforeDrawing", fn)
+
+    def onAfterDrawing(fn: (Canvas) => Unit) = underlying.on("afterDrawing", fn)
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l278)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 25]
```diff
@@ -0,0 +1,282 @@
...
+package object vis {
+
+  @js.native
+  trait Node extends js.Any {
+    var id: js.Any = js.native
+
+  }
+
+  object Node {
+    def apply(id: String, label: String, title: js.UndefOr[String] = js.undefined, value: js.UndefOr[Int] = js.undefined, color: js.UndefOr[String] = js.undefined): vis.Node = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [File line length exceeds 160 characters](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l19)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 26]
```diff
@@ -0,0 +1,282 @@
...
+
+    def onContext(fn: ClickEvent => Unit) = underlying.on("onContext", fn)
+
+    def onHold(fn: ClickEvent => Unit) = underlying.on("hold", fn)
+
+    def onRelease(fn: ClickEvent => Unit) = underlying.on("release", fn)
+
+    def onSelect(fn: ClickEvent => Unit) = underlying.on("select", fn)
+
+    def onSelectNode(fn: ClickEvent => Unit) = underlying.on("selectNode", fn)
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l245)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 27]
```diff
@@ -0,0 +1,282 @@
...
+
+    def onSelectEdge(fn: ClickEvent => Unit) = underlying.on("selectEdge", fn)
+
+    def onDeselectNode(fn: DeselectEvent => Unit) = underlying.on("deselectNode", fn)
+
+    def onDeselectEdge(fn: DeselectEvent => Unit) = underlying.on("deselectEdge", fn)
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l251)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 28]
```diff
@@ -0,0 +1,282 @@
...
+  }
+
+  object Edge {
+    def apply(id: String, from: String, to: String, arrows: js.UndefOr[String] = "to", title: js.UndefOr[String] = js.undefined, value: js.UndefOr[Int] = js.undefined): vis.Edge = {
+      js.Dynamic.literal(id = id, from = from, to = to, value = value, arrows = arrows).asInstanceOf[vis.Edge]
+    }
+  }
+
+  @js.native
+  trait DataSetOptions extends js.Any {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Redundant braces after class definition](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l36)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 29]
```diff
@@ -0,0 +1,282 @@
...
+
+  @js.native
+  trait ClickEvent extends js.Any with Selection {
+    var event: Event = js.native
+    var pointer: PointerCoords = js.native
+  }
+
+  @js.native
+  trait Selection extends js.Any {
+    var nodes: js.Array[String] = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l201)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 30]
```diff
@@ -0,0 +1,282 @@
...
+  }
+
+  @js.native
+  trait EdgeEvent extends js.Any {
+    var edge: String = js.native
+  }
+
+  @js.native
+  trait NodeEvent extends js.Any {
+    var node: String = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2160736326/issues/source?bid=3121282&fileBranchId=3158877#l223)

File: frontend/src/main/scala/akkaviz/frontend/ActorRepository.scala [hunk 31]
```diff
@@ -13,7 +13,7 @@ class ActorRepository {
 
   val seenActors = Var[Set[String]](Set())
 
-  def state(actor: String): Var[ActorState] = currentActorState.getOrElseUpdate(actor, Var(ActorState(actor, FrontendUtil.shortActorName(actor))))
+  def state(actor: String): Var[ActorState] = currentActorState.getOrElseUpdate(actor, Var(ActorState(actor, FrontendUtil.shortActorName(actor), FrontendUtil.systemName(actor))))
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ActorRepository.scala]: Codacy Issue found: [File line length exceeds 160 characters](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074159/issues/source?bid=3121282&fileBranchId=3158877#l16)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 32]
```diff
@@ -1,100 +1,124 @@
...
-  }
+  private[this] val currentlyVisibleNodes = js.Dictionary[Unit]()
 
-  private val graph = DOMGlobalScope.graph
-  private val nodes = js.Dictionary[js.Dictionary[js.Any]]()
-  private val connectedNodes = js.Dictionary[Unit]()
-  private val createdLinks = js.Dictionary[Unit]()
+  private[this] val networkNodes = new vis.DataSet[vis.Node]()
+  private[this] val networkEdges = new vis.DataSet[vis.Edge]()
+  private[this] var network: js.UndefOr[vis.Network] = js.undefined
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074171/issues/source?bid=3121282&fileBranchId=3158877#l23)

File: frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala [hunk 33]
```diff
@@ -1,100 +1,124 @@
...
 import rx.Var
 
-import scala.concurrent.duration.{FiniteDuration, _}
 import scala.scalajs.js
-import scala.scalajs.js.timers.SetTimeoutHandle
+import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.|._
 
-class GraphView(showUnconnected: Var[Boolean]) extends Component {
+class GraphView(showUnconnected: Var[Boolean], actorSelectionToggler: (String) => Unit, renderer: (String, ActorState) => vis.Node) extends Component with GraphViewSettings {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/GraphView.scala]: Codacy Issue found: [File line length exceeds 160 characters](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074171/issues/source?bid=3121282&fileBranchId=3158877#l13)

File: frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074167/issues/source?bid=3121282&fileBranchId=3158877#l19)

File: frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/ScheduledQueue.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074167/issues/source?bid=3121282&fileBranchId=3158877#l8)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 34]
```diff
@@ -0,0 +1,286 @@
...
+    //Methods to get information on nodes and edges.
+
+    //Physics methods to control when the simulation should run.
+
+    //Selection methods for nodes and edges.
+    def getSelection(): Selection = js.native
+
+    def getSelectedNodes(): js.Array[String] = js.native
+
+    def getSelectedEdges(): js.Array[String] = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074181/issues/source?bid=3121282&fileBranchId=3158877#l162)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 35]
```diff
@@ -0,0 +1,286 @@
...
+    def setSize(width: String, height: String): Unit = js.native
+
+    //Methods to get information on nodes and edges.
+
+    //Physics methods to control when the simulation should run.
+
+    //Selection methods for nodes and edges.
+    def getSelection(): Selection = js.native
+
+    def getSelectedNodes(): js.Array[String] = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074181/issues/source?bid=3121282&fileBranchId=3158877#l160)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 36]
```diff
@@ -0,0 +1,286 @@
...
+
+    def getSelectedEdges(): js.Array[String] = js.native
+
+    def getNodeAt(domPosition: Coords): String = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074181/issues/source?bid=3121282&fileBranchId=3158877#l164)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 37]
```diff
@@ -0,0 +1,286 @@
...
+    def once(event: String, fn: js.Function): Unit = js.native
+
+    //Methods related to the canvas
+    def canvasToDOM(coords: Coords): Coords = js.native
+
+    def DOMtoCanvas(coords: Coords): Coords = js.native
+
+    def redraw(): Unit = js.native
+
+    def setSize(width: String, height: String): Unit = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074181/issues/source?bid=3121282&fileBranchId=3158877#l151)

File: frontend/src/main/scala/akkaviz/frontend/vis/package.scala [hunk 38]
```diff
@@ -0,0 +1,286 @@
...
+    def selectNodes(nodeIds: js.Array[String], highlightEdges: Boolean = ???): Unit = js.native
+
+    def selectEdges(edgeIds: js.Array[String]): Unit = js.native
+
+    def setSellection(selection: Selection, options: js.UndefOr[SelectionOptions] = js.undefined): Unit = js.native
+
+    def unselectAll(): Unit = js.native
+
+    //Methods to control the viewport for zoom and animation.
+    def getScale(): Double = js.native
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/vis/package.scala]: Codacy Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2162074181/issues/source?bid=3121282&fileBranchId=3158877#l177)

File: monitoring/src/main/resources/web/index.html [hunk 39]
```diff
@@ -40,8 +41,11 @@
         <div class="tab-pane fade panel panel-default" id="messagefiltering"></div>
         <div class="tab-pane fade panel panel-default" id="globalsettings">
             <div id="onoffsettings" class="panel panel-default"></div>
-            <div class="panel panel-default"><div id="graphsettings" class="panel-body"></div></div>
+            <div class="panel panel-default">
+                <div id="graphsettings" class="panel-body"></div>
+            </div>
             <div id="receivedelay" class="panel panel-default"></div>
+            <div id="graphsettings" class="panel panel-defaul"></div>
```
Comment (JJag) [inline monitoring/src/main/resources/web/index.html]: Duplicated id

**Pull request merged**